### PR TITLE
Improve rendering of image type inside thumbs

### PR DESCRIPTION
**Pull request by grand-piano**
The display of the image type inside thumbs has two issues. 

1. The font is a bit too large. 
2. Characters with descenders are overwritten by the image in landscape format. 
2. For images in portrait format, only the first letter of the image type is readable.

This is what I've changed:
- The font size has been slightly reduced
- Tie image type is displayed uniformly in upper case
- For portrait images, the image type is written vertically

The fix is not perfect, because depending on the aspect ratio the text is still partially overwritten, but the fix is a improvement of the current state. 

**darktable 2.4.4**

darktablethumb

**Fix**

thumbfix

### CODE REVIEW

**Comment by TurboGit** [comment]
Sounds like a nice enhancement to me.

File: src/views/view.c [hunk 1]
```diff
@@ -915,18 +915,45 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
...
       pango_layout_set_font_description(layout, desc);
       const char *ext = img->filename + strlen(img->filename);
       while(ext > img->filename && *ext != '.') ext--;
       ext++;
       cairo_set_source_rgb(cr, fontcol, fontcol, fontcol);
-      pango_layout_set_text(layout, ext, -1);
-      pango_layout_get_pixel_extents(layout, &ink, NULL);
-      cairo_move_to(cr, .025 * width - ink.x, .24 * height - fontsize);
-      pango_cairo_show_layout(cr, layout);
+
```
Comment (TurboGit) [inline src/views/view.c]: superfluous empty line

File: src/views/view.c [hunk 2]
```diff
@@ -915,18 +915,45 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
...
+
+      const int isportrait = ((img->orientation & ORIENTATION_SWAP_XY) != 0) != (img->height > img->width);
+      char* upcase_ext = g_ascii_strup(ext, -1);  // extension in capital letters to avoid character descenders
+
+      if ( isportrait )
```
Comment (TurboGit) [inline src/views/view.c]: style is no space after ( and before )

File: src/views/view.c [hunk 3]
```diff
@@ -915,18 +915,45 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
...
       while(ext > img->filename && *ext != '.') ext--;
       ext++;
       cairo_set_source_rgb(cr, fontcol, fontcol, fontcol);
-      pango_layout_set_text(layout, ext, -1);
-      pango_layout_get_pixel_extents(layout, &ink, NULL);
-      cairo_move_to(cr, .025 * width - ink.x, .24 * height - fontsize);
-      pango_cairo_show_layout(cr, layout);
+
+
+      const int isportrait = ((img->orientation & ORIENTATION_SWAP_XY) != 0) != (img->height > img->width);
```
Comment (TurboGit) [inline src/views/view.c]: this is not correct as it checks the original orientation. take a landscape picture, use crop&rotate and create a portrait composition. You'll see that the format will be hidden under the picture. You should use the height and width as cropped.

Comment (grand-piano) [reply src/views/view.c]: Thanks, you're right.
I've been trying to figure out img->crop_width and img->crop_height, but both are always 0. My new commit works for me.

**Comment by TurboGit** [comment]
All good now. Thanks.

**Pull request merged**